packages/terraform-modules-public/testnet/modules/validator/startup.sh

```diff
@@ -0,0 +1,193 @@
+#!/bin/bash
+
+# ---- Set Up Logging ----
+
+curl -sSO https://dl.google.com/cloudagents/install-logging-agent.sh
+bash install-logging-agent.sh
+
+# ---- Set Up Persistent Disk ----
+
+# gives a path similar to `/dev/sdb`
+DISK_PATH=`readlink -f /dev/disk/by-id/google-${attached_disk_name}`
+DATA_DIR=/root/.celo
+
+echo "Setting up persistent disk ${attached_disk_name} at $DISK_PATH..."
+
+DISK_FORMAT=ext4
+CURRENT_DISK_FORMAT=`lsblk -i -n -o fstype $DISK_PATH`
+
+echo "Checking if disk $DISK_PATH format $CURRENT_DISK_FORMAT matches desired $DISK_FORMAT..."
+
+# If the disk has already been formatted previously (this will happen
+# if this instance has been recreated with the same disk), we skip formatting
+if [[ $CURRENT_DISK_FORMAT == $DISK_FORMAT ]]; then
+  echo "Disk $DISK_PATH is correctly formatted as $DISK_FORMAT"
+else
+  echo "Disk $DISK_PATH is not formatted correctly, formatting as $DISK_FORMAT..."
+  mkfs.ext4 -m 0 -F -E lazy_itable_init=0,lazy_journal_init=0,discard $DISK_PATH
+fi
+
+echo "Mounting $DISK_PATH onto $DATA_DIR"
+mkdir -p $DATA_DIR
+mount -o discard,defaults $DISK_PATH $DATA_DIR
+[[ ${reset_geth_data} == "true" ]] && rm -rf $DATA_DIR/geth
+mkdir -p $DATA_DIR/account
+
+# ---- Install Docker ----
+
+echo "Installing Docker..."
+apt update && apt upgrade
+apt install -y apt-transport-https ca-certificates curl software-properties-common gnupg2
+curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add -
+add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/debian $(lsb_release -cs) stable"
+apt update && apt upgrade
+apt install -y docker-ce
+systemctl start docker
+
+echo "Configuring Docker..."
+cat <<'EOF' > '/etc/docker/daemon.json'
+{
```

Comment:
is it worth splitting out these configuration files and hosting them in separate self-contained files?

If you do that, I'd re-use them for configuring Debian with Azure Templates.

Reply:
...actually, most of this file is reusable for the Azure debian setup, with some differences on how sensitive data (e.g., `validator_private_key`) is handled.  Worth splitting the whole `startup.sh` file out into a separate PR and location (e.g., `packages/debian` or something)?

Reply:
Yeah, that makes a lot of sense. I was thinking about the AWS module and most of the script will be reusable also for it, so we can think about organizing the code so it can be easily imported for each component.

Reply:
cool -- do you have cycles to do that?  do you want me to help with moving the `startup.sh`s elsewhere?

Reply:
Hi Silas,
I was thinking about merging this PR and then address that in a new PR, so we can close this PR asap and do not postpone too far. What do you think?

Reply:
:+1: